core/core_builders.py

```diff
@@ -1,12 +1,15 @@
 """Functions used to generate source files during build time"""
 
 from collections import OrderedDict
-from io import TextIOWrapper
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from io import TextIOWrapper
```

Comment:
since it is only used for type checking there is no need to import it during run